mlflow/metrics/metric_definitions.py

```diff
@@ -275,11 +275,28 @@ def _mse_eval_fn(predictions, targets=None, metrics=None, sample_weight=None):
         return MetricValue(aggregate_results={"mean_squared_error": mse})
 
 
-def _rmse_eval_fn(predictions, targets=None, metrics=None, sample_weight=None):
-    if targets is not None and len(targets) != 0:
+def _root_mean_squared_error(*, y_true, y_pred, sample_weight):
+    try:
         from sklearn.metrics import root_mean_squared_error
+    except ImportError:
+        # If root_mean_squared_error is unavailable, fall back to
+        # `mean_squared_error(..., squared=False)`, which is deprecated in scikit-learn >= 1.4.
+        pass
```

Comment:
nit: Can we put `return mean_squared_error(...)` here? The `try-except-else` syntax is a bit less readable.